src/index.ts

```diff
@@ -14,6 +15,7 @@ appInsights.setup(clientKey)
   .setUseDiskRetryCaching(true)
   .setSendLiveMetrics(process.env.AI_ENABLE_LIVEMETRICS === 'true') // Default to disabled: https://github.com/microsoft/ApplicationInsights-node.js/issues/615
   .setDistributedTracingMode(appInsights.DistributedTracingModes.AI_AND_W3C)
+appInsights.defaultClient.config.httpsAgent = new Agent({ keepAlive: true, maxSockets: 400 })
```

Comment:
maxSockets defaults to infinity.  I don't think you need to override it.
maxFreeSockets defines the size of idle pool.  It defaults to 256, and is way too high IMHO.  We should set maxFreeSockets to 10 at most.

Reply:
Josh what do you think is a sensible value for maxSockets?

Reply:
set to 20